tests/full_node_test.cpp

```diff
@@ -714,12 +714,18 @@ TEST_F(FullNodeTest, sync_five_nodes) {
   context.assert_all_transactions_known();
   context.assert_balances_synced();
 
-  // Prune state_db of one node to remove all but last 10 block state and verify all ballances still match
+  // Prune state_db of one node
   auto prune_node = nodes[nodes.size() - 1];
-  auto last_block = prune_node->getPbftChain()->getPbftChainSize() - 10;
-  prune_node->getFinalChain()->prune(last_block);
-  context.assert_all_transactions_success();
-  context.assert_all_transactions_known();
+  const uint32_t min_blocks_to_prune = 50;
+  // This ensures that we never prune blocks that are over proposal period
+  ASSERT_HAPPENS({100s, 100ms}, [&](auto &ctx) {
```

Comment:
Is it normal that this test could possibly run for 100s? That is very big value and we should set as low value as possible for this waits imo

Reply:
Yes, I put that while I was testing it. It should not be more than 10s. I will replace it with 20s